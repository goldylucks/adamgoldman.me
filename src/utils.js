import cloudinary from 'cloudinary-core'

/* eslint-disable no-extend-native,func-names */
String.prototype.capitalize = function () {
  return this[0].toUpperCase() + this.substr(1).toLowerCase()
}

Array.prototype.last = function () {
  return this[this.length - 1]
}
/* eslint-enable no-extend-native */

const cloudinaryCore = new cloudinary.Cloudinary({ cloud_name: 'goldylucks' })

export const cloudImg = imgName => cloudinaryCore.url(imgName)

export const noop = () => null

export const isProd = process.env.NODE_ENV === 'production'

export const scrollToElem = (element, to, duration) => {
  if (duration <= 0) return
  const difference = to - element.scrollTop
  const perTick = difference / duration * 10

  setTimeout(() => {
    // eslint-disable-next-line operator-assignment
    element.scrollTop = element.scrollTop + perTick
    if (element.scrollTop === to) return
    scrollToElem(element, to, duration - 10)
  }, 10)
}

export const filterDrafts = item => !item.isDraft

export const titleToSlug = title => title.toLowerCase().replace(/ /g, '-')

export const getSlug = (item) => {
  if (item.type === 'pages') {
    return `/${item.url}`
  }
  if (item.type === 'tags') {
    return `/tags/${titleToSlug(item.title)}`
  }
  return `/${item.type}/${item.url}`
}

const parseFbMsg = ({
  type, html, alt, src, style, duration,
}) => {
  if (type === 'voiceMsg') {
    return `Voice msg - Duration: ${duration}`
  }

  if (type === 'sticker') {
    return `<div style="${style}"></div>`
  }

  if (type === 'emoticon' || type === 'image') {
    return `<img alt=${alt} src=${src} />`
  }

  if (type === 'likeSticker') {
    return '{LIKE}'
  }

  if (type === 'textWithHtml') {
    return html
  }

  if (type === 'textWithEmoticon') {
    return html
  }

  if (type === 'text') {
    return html
  }
  console.error('Caught message with no type, that should not happen!') // eslint-disable-line no-console
  return ''
}

export const parseFbConversation = arrOfMsgs =>
  arrOfMsgs
    .map(({ author, content }) => {
      if (author === 'time') {
        return `<div class="chat-message-time">${content}</div>`
      }
      return content
        .map(msg =>
          `<div class="chat-message-container clearfix ${author} ${msg.isRtl
            ? 'rtl'
            : ''}"><div class="chat-message">${parseFbMsg(msg)}</div></div>`)
        .join('')
    })
    .join('')

export const comment = md => ({ author: 'comment', md })

export const reorder = (list, startIndex, endIndex) => {
  const result = Array.from(list)
  const [removed] = result.splice(startIndex, 1)
  result.splice(endIndex, 0, removed)

  return result
}

export const isMobile = () => window.innerWidth <= 800

export const isSavoring = path => path.includes('savoring-your-child')
