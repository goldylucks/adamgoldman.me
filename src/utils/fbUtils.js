import axios from 'axios'

import { DOMAIN, FB_APP_ID } from '../constants'

export const fbLinkByUserId = id => `https://www.facebook.com/profile.php?id${id}`

export const fbAuth = (cb) => {
  global.FB.login(res => fbAuth2(res, cb), { scope: 'email,public_profile' })
}

export const getFbShareUrl = path => `https://www.facebook.com/sharer/sharer.php?u=${DOMAIN}${path}`

/* eslint-disable */
export const initFbSdk = () => {
  window.fbAsyncInit = function() {
    FB.init({
      appId            : FB_APP_ID,
      autoLogAppEvents : true,
      xfbml            : true,
      version          : 'v2.11'
    });
  };

  (function(d, s, id){
     var js, fjs = d.getElementsByTagName(s)[0];
     if (d.getElementById(id)) {return;}
     js = d.createElement(s); js.id = id;
     js.src = "https://connect.facebook.net/en_US/sdk.js";
     fjs.parentNode.insertBefore(js, fjs);
   }(document, 'script', 'facebook-jssdk'));
};
/* eslint-enable */

function fbAuth2(response, cb) {
  if (response.status !== 'connected') {
    global.alert('please authorize login to continue')
    return
  }
  const { accessToken, userID } = response.authResponse
  global.FB.api('/me?fields=id,name,picture,gender',
    ({ name, picture, gender }) => {
      userServerLogin({
        fbUserId: userID, fbClientAccessToken: accessToken, name, picture, gender,
      }, cb)
    },
  )
}

function userServerLogin(user, cb) {
  if (user.picture && user.picture.data && user.picture.data.url) {
    user.fbPictureUrl = user.picture.data.url
  }
  axios.post('/api/users/fbAuth', user)
    .then((serverRes) => { cb(serverRes.data) })
    .catch((err) => {
      global.console.error(err)
      global.alert('there was an error, please contact me')
    })
}
