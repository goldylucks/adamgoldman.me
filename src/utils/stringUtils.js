// eslint-disable-next-line import/prefer-default-export
export const replaceAtIdx = (str, idx, replacement) => str.substr(0, idx)
 + replacement + str.substr(idx + 1)
