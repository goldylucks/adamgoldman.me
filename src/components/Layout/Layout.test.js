import React from 'react'
import renderer from 'react-test-renderer'

import { initFbSdk } from '../../utils/fbUtils'
import Wrapper from '../../../test/TestWrapper'

describe('Layout', () => {
  test('renders children correctly', () => {
    const wrapper = renderer
      .create(
        <Wrapper>
          <div className="child" />
        </Wrapper>,
      )
      .toJSON()

    expect(wrapper).toMatchSnapshot()
    expect(initFbSdk).toHaveBeenCalled()
  })
})
