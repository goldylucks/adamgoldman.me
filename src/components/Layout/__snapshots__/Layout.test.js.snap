// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders children correctly 1`] = `
<div
  className="container"
>
  <nav
    className="navbar navbar-expand-lg fixed-top main-nav navbar-light"
  >
    <div
      className="container"
    >
      <div
        className="clearfix d-sm-block d-md-none"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-bars fa-w-14 "
          data-icon="bars"
          data-prefix="fas"
          onClick={[Function]}
          role="img"
          style={
            Object {
              "marginRight": 10,
            }
          }
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <div
          className="mobileMenu"
        >
          <nav>
            <h3
              className="mobileMenuHeadline"
              onClick={[Function]}
            >
              Adam Goldman
            </h3>
            <ul
              className="navbar-nav"
            >
              <li
                className="nav-item"
              >
                <a
                  className="nav-link navLink"
                  href="/"
                  onClick={[Function]}
                >
                  Home
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link navLink"
                  href="/transcripts"
                  onClick={[Function]}
                >
                  Transcripts
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link navLink"
                  href="/tools"
                  onClick={[Function]}
                >
                  Tutorials
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link navLink"
                  href="/about"
                  onClick={[Function]}
                >
                  About
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link navLink"
                  href="https://www.adamgoldman.me/contact"
                  rel="nofollow noreferrer noopener"
                  target="_blank"
                >
                  Contact
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link navLink"
                  href="/i-dont-charge-i-accept"
                  onClick={[Function]}
                >
                  <span>
                    Pricing 
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-heart fa-w-16 "
                      data-icon="heart"
                      data-prefix="fas"
                      role="img"
                      style={
                        Object {
                          "color": "red",
                          "marginLeft": 5,
                        }
                      }
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </span>
                </a>
              </li>
              <li
                className="nav-item"
                style={
                  Object {
                    "marginTop": 20,
                  }
                }
              >
                <a
                  className="nav-link btn btn-primary btn-block"
                  href="https://www.adamgoldman.me/book-session"
                  rel="nofollow noreferrer noopener"
                  target="_blank"
                >
                  Book a session
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
      <a
        className="navbar-brand mr-auto"
        href="/"
        onClick={[Function]}
      >
        Adam Goldman
      </a>
      <div
        className="collapse navbar-collapse"
      >
        <ul
          className="navbar-nav ml-auto"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/transcripts"
              onClick={[Function]}
            >
              Transcripts
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/tools"
              onClick={[Function]}
            >
              Tutorials
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/about"
              onClick={[Function]}
            >
              About
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www.adamgoldman.me/contact"
              rel="nofollow noreferrer noopener"
              target="_blank"
            >
              Contact
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/i-dont-charge-i-accept"
              onClick={[Function]}
            >
              <span>
                Pricing 
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-heart fa-w-16 "
                  data-icon="heart"
                  data-prefix="fas"
                  role="img"
                  style={
                    Object {
                      "color": "red",
                      "marginLeft": 5,
                    }
                  }
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </span>
            </a>
          </li>
          <li
            className="nav-item"
            style={
              Object {
                "marginLeft": 10,
              }
            }
          >
            <a
              className="nav-link btn btn-primary btn-sm"
              href="https://www.adamgoldman.me/book-session"
              rel="nofollow noreferrer noopener"
              target="_blank"
            >
              Book a session
            </a>
          </li>
        </ul>
      </div>
      <a
        className="nav-link btn btn-primary btn-sm d-sm-block d-md-none"
        href="https://www.adamgoldman.me/book-session"
        rel="nofollow noreferrer noopener"
        target="_blank"
      >
        Book a session
      </a>
    </div>
  </nav>
  <div
    className="innerContainer"
  >
    <div
      className="child"
      onLogin={[Function]}
      onStartToolResponse={[Function]}
      onSubmitIntro={undefined}
      onSubmitModule={[Function]}
      user={Object {}}
    />
  </div>
  <div
    className="container container-footer"
  >
    <div
      className="footer"
    >
      <div
        className="row"
      >
        <div
          className="col-sm-6"
        >
          <p
            className="pull-left"
          >
            Copyright Â© 2017 
            <a
              className="link"
              href="/"
              onClick={[Function]}
            >
              Adam Goldman
            </a>
          </p>
        </div>
        <div
          className="col-sm-6"
        >
          <p
            className="pull-right"
          >
            <a
              className="link"
              href="/legal-stuff"
              onClick={[Function]}
            >
              legal stuff
            </a>
            <a
              className="link"
              href="/privacy"
              onClick={[Function]}
              style={
                Object {
                  "marginLeft": 10,
                }
              }
            >
              Privacy
            </a>
          </p>
        </div>
      </div>
      <div
        className="clearfix"
      />
    </div>
  </div>
  <a
    className="container"
    href="https://www.adamgoldman.me/contact"
    rel="nofollow noreferrer noopener"
    target="_blank"
  >
    <img
      alt="Messenger Link"
      src="http://res.cloudinary.com/goldylucks/image/upload/v1/adamgoldman.me/messenger"
    />
  </a>
</div>
`;
